### PR TITLE
Dialect callback to resolve SqlTypeDescriptor from JDBC type code, precision and scale

### PR DESCRIPTION
**Pull request by beikov**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/dialect/TeradataDialect.java [hunk 1]
```diff
@@ -120,6 +122,24 @@ public TeradataDialect(int version) {
...
 
+	@Override
+	public SqlTypeDescriptor resolveSqlTypeDescriptor(
+			int jdbcTypeCode,
+			int precision,
+			int scale,
+			SqlTypeDescriptorRegistry sqlTypeDescriptorRegistry) {
+		switch ( jdbcTypeCode ) {
+			case Types.BIT:
+				return sqlTypeDescriptorRegistry.getDescriptor( Types.BOOLEAN );
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/dialect/TeradataDialect.java:133]: Can't we get rid of stuff like this now?

File: hibernate-core/src/main/java/org/hibernate/dialect/TimesTenDialect.java [hunk 2]
```diff
@@ -95,6 +97,18 @@ public int getVersion() {
...
 	}
 
+	@Override
+	public SqlTypeDescriptor resolveSqlTypeDescriptor(
+			int jdbcTypeCode,
+			int precision,
+			int scale,
+			SqlTypeDescriptorRegistry sqlTypeDescriptorRegistry) {
+		if ( jdbcTypeCode == Types.BIT ) {
+			return sqlTypeDescriptorRegistry.getDescriptor( Types.BOOLEAN );
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/dialect/TimesTenDialect.java:107]: This too.

File: hibernate-core/src/main/java/org/hibernate/dialect/TeradataDialect.java (hunk 1 above)
Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/dialect/TeradataDialect.java:133]: I don't think we can because this is for supporting native queries.

Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/dialect/TeradataDialect.java:133]: Oh it's for reflecting on the metadata of the native query JDBC result set and guessing the Hibernate types?

OK, that makes sense.

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/dialect/TeradataDialect.java:133]: Correct

**Review by sebersole — APPROVED** [review]
LGTM

**Pull request merged**